Src/WitsmlExplorer.Api/Workers/AnalyzeGapWorker.cs

```diff
@@ -0,0 +1,173 @@
+                .ToList();
+            
+            Index gapSize = isDepthLog && logCurveMinMaxIndex.MinIndex is DepthIndex depthMinIndex
+                ? new DepthIndex(job.GapSize, depthMinIndex.Uom)
+                : new TimeSpanIndex(job.TimeGapSize);
+
+            gapReportItems.AddRange(GetAnalyzeGapReportItem(logMnemonic.value, inputAnalyzeDataList, gapSize, isLogIncreasing));
+        }
+
+        return GetGapReportResult(job, gapReportItems, isDepthLog, logUid);
+    }
+
+    private static Func<Index, bool> ValidateCurveIndex(bool isLogIncreasing, LogCurveIndex logCurveIndex)
+    {
+        return isLogIncreasing
+            ? value => value >= logCurveIndex.MinIndex && value <= logCurveIndex.MaxIndex
+            : value => value <= logCurveIndex.MinIndex && value >= logCurveIndex.MaxIndex;
+    }
+
+    private static Index GetDepthOrDateTimeIndex(bool isDepthLog, string value, LogCurveIndex logCurveIndex)
+    {
+        return isDepthLog && logCurveIndex.MinIndex is DepthIndex depthMinIndex
+            ? new DepthIndex(double.Parse(value, CultureInfo.InvariantCulture), (depthMinIndex).Uom)
+            : new DateTimeIndex(DateTime.Parse(value, CultureInfo.InvariantCulture));
+    }
+
+    private (WorkerResult, RefreshAction) GetGapReportResult(AnalyzeGapJob job, List<AnalyzeGapReportItem> gapReportItems, bool isDepth, string logUid)
+    {
+        Logger.LogInformation("Analyzing gaps is done. {jobDescription}", job.Description());
+        job.JobInfo.Report = GetGapReport(gapReportItems, job.LogReference, isDepth);
+        WorkerResult workerResult = new(GetTargetWitsmlClientOrThrow().GetServerHostname(), true, $"Analyze gaps for log: {logUid}", jobId: job.JobInfo.Id);
+        return (workerResult, null);
+    }
+
+    private AnalyzeGapReport GetGapReport(List<AnalyzeGapReportItem> analyzeGapItems, LogObject logReference, bool isDepthLog)
+    {
+        return new AnalyzeGapReport
+        {
+            Title = $"Analyze gaps report",
+            Summary = analyzeGapItems.Count > 0
+                ? $"Found {analyzeGapItems.Count} gaps for {(isDepthLog ? "depth" : "time")} log '{logReference.Name}':"
+                : "No gaps were found for selected items.",
+            LogReference = logReference,
+            ReportItems = analyzeGapItems
+        };
+    }
+
+    /// <summary>
+    /// Get all gaps for input list of mnemonic values larger then requested gap size.
+    /// </summary>
+    /// <param name="mnemonic">Name of mnemonic.</param>
+    /// <param name="inputIndexList">Input list of indexes.</param>
+    /// <param name="requestedGapSize">Requested gap size defined by user.</param>
+    /// <param name="isLogIncreasing">Is log direction increasing.</param>
+    /// <returns>Report items with gap size information.</returns>
+    private IEnumerable<AnalyzeGapReportItem> GetAnalyzeGapReportItem(string mnemonic, IList<Index> inputIndexList, Index requestedGapSize, bool isLogIncreasing)
```

Comment:
It seems to not be working for decreasing logs. Tested with a gap size smaller than the step between each index, but still got no results back.

Reply:
It should work for decreasing logs it depends also on setting min or max index of curve. I added unit tests for decreasing direction.